client/src/components/primitiveComponents/AlertBoxes.tsx

```diff
@@ -3,30 +3,87 @@ import { AlertDialog, Button, Flex, TextField } from '@radix-ui/themes';
 import { useState } from 'react';
 import { TwilightSelect } from './Primitives';
 
-interface TLAddTimelineAlertProps {
-    year: number;
-    onSubmit: (year: number, timelineEvent: string) => void;
+interface TwilightEditTextAlertProps {
+    labelText: string;
+    textInput: string;
+    objectKey?: string;
+    onSubmit: (newValue: string, objectKey?: string) => void;
+    onDelete?: () => void;
+}
+
+const TwilightEditTextAlert = ({ labelText, textInput, objectKey, onSubmit, onDelete }: TwilightEditTextAlertProps) => {
+    const [displayedText, setDisplayedText] = useState(textInput);
+
+    const handleSubmitEntry = () => {
+        if (objectKey !== undefined) {
+            onSubmit(displayedText, objectKey);
+        } else {
+            onSubmit(displayedText);
+        }
```

Comment:
I could be mistaken, but I don't think you need this if/else
I believe you can just call onSubmit(displayedText, objectKey) and objectKey will be ignored if it's undefined.
Idk test and find out

Reply:
it seems to be working